cmd/mirror-url.go

```diff
@@ -136,7 +136,7 @@ func deltaSourceTarget(sourceURL, targetURL string, isFake, isOverwrite, isRemov
 			// No difference, continue.
 		case differInType:
 			URLsCh <- URLs{Error: errInvalidTarget(diffMsg.SecondURL)}
-		case differInSize, differInTime:
+		case differInSize, differInSrcTime, differInTgtTime:
```

Comment:
We only need to handle differInSrcTime - we simply ignore differInTgtTime.

Reply:
We need to take care of `differInTgtTime` for the case when `tgtTime` is newer than `srcTime`.
See below code, between line numbers 204~213 in difference.go file:

```
:
:
} else if tgtTime.After(srcTime) {
	// Regular files differing in timestamp.
	// Target timestamp is newer.
	diffCh <- diffMessage{
		FirstURL:      srcCtnt.URL.String(),
		SecondURL:     tgtCtnt.URL.String(),
		Diff:          differInTgtTime,
		firstContent:  srcCtnt,
		secondContent: tgtCtnt,
	}
}
:
:
```

Otherwise, we fall into `case default:` where we see the following error message:

```
$ ./mc mirror myminio/dir/ myminio/mybucketnew/
 0 B / ? ┃░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▓┃
srcTime 2018-01-30 21:41:00.906 +0000 UTC
tgtTime 2018-01-30 22:02:36.11 +0000 UTC
mc: <ERROR> Unable to prepare URL for copying. Unrecognized diffType: time provided.
```

Reply:
We don't need to do that target is assumed to be always same as source, if target is modified we shouldn't do anything about it @ebozduman  and it is treated as unsupported. 

NOTE: `mc mirror` is uni-directional i.e all decisions made to copy data are done only source to target. What we need to do is simply ignore the case of `differInTgtTime`  not even go to `default` case.

Reply:
How about if user still would like to overwrite when target file is newer than the source file?

Reply:
I talked to @harshavardhana .
`"mc mirror"` is not going to do anything when source file modification time is behind target file modification time, which should never happen and if it does, it is all user's responsibility how to handle it. This is exactly the same behavior with s3 sync command.
"mc mirror" output in this case will be like:
```
$ ./mc mirror myminio/mybucket myminio/dir --overwrite
 0 B / ? ┃░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▓┃ 0s
```